src/Ads/AccountService.php

```diff
@@ -215,46 +216,52 @@ public function get_billing_status(): array {
 	/**
 	 * Gets the ads account access status.
 	 *
-	 * @return bool
+	 * @return array
 	 * @throws Exception When email is not present or invite link is empty when email present and has no account access.
 	 */
 	public function get_ads_account_has_access() {
-		try {
-			// Check if ads id is present.
-			$ads_id = $this->options->get_ads_id();
-
-			if ( empty( $ads_id ) ) {
-				return [
-					'has_access'  => false,
-					'invite_link' => '',
-				];
-			}
+		// Check if ads id is present.
+		$ads_id = $this->options->get_ads_id();
 
-			$connection_status  = $this->container->get( Connection::class )->get_status();
-			$email              = $connection_status['email'] ?? '';
-			$accept_invite_link = $this->options->get( OptionsInterface::ADS_BILLING_URL, '' );
+		if ( empty( $ads_id ) ) {
+			throw new Exception( __( 'Ads id not present', 'google-listings-and-ads' ) );
+		}
 
-			// If no email, means google account is not connected.
-			if ( empty( $email ) ) {
-				throw new Exception( 'Google account is not connected' );
-			}
+		$connection_status  = $this->container->get( Connection::class )->get_status();
+		$email              = $connection_status['email'] ?? '';
+		$accept_invite_link = $this->options->get( OptionsInterface::ADS_BILLING_URL, '' );
 
-			$status = $this->container->get( Ads::class )->has_access( $email );
-			$status = boolval( $status );
+		// If no email, means google account is not connected.
+		if ( empty( $email ) ) {
+			throw new Exception( __( 'Google account is not connected', 'google-listings-and-ads' ) );
+		}
 
-			// If no access to ads account and there is no invite_link, throw exception.
-			if ( ! $status && empty( $accept_invite_link ) ) {
-				throw new Exception( 'Billing URL is not present' );
-			}
+		$status = $this->container->get( Ads::class )->has_access( $email );
+
+		// If no access to ads account and there is no invite_link, throw exception.
+		if ( ! $status && empty( $accept_invite_link ) ) {
+			throw new Exception( __( 'Billing URL is not present', 'google-listings-and-ads' ) );
+		}
```

Comment:
I think it's possible that for Ads accounts created outside the plugin, we would not have an invite link, so I don't think we will want to throw an exception here. Returning an empty value will probably be fine, and then the front end can choose what a fallback URL should be (if any) in case we want to direct the user to more information about how to accept their account.

Reply:
This makes sense. Updated.